Sources/MapboxNavigation/InstructionPresenter.swift

```diff
@@ -292,12 +313,8 @@ class InstructionPresenter {
         return NSAttributedString(attachment: attachment)
     }
     
-    private func completeShieldDownload(_ image: UIImage?) {
-        guard image != nil else { return }
-        // We *must* be on main thread here, because attributedText() looks at object properties only accessible on main thread.
-        DispatchQueue.main.async {
-            self.onShieldDownload?(self.attributedText()) //FIXME: Can we work with the image directly?
-        }
+    private func completeShieldDownload() {
+        self.onShieldDownload?(self.attributedText())
```

Comment:
```suggestion
        onShieldDownload?(attributedText())
```

Reply:
Thanks for the suggestion, I'll update this part and the documentation above.